src/Components/Classes/Table/Table.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { useState } from 'react';
+
+const Table = ({ data, deleteItem, onEditItem, activities, trainers }) => {
+  const [klass, setKlass] = useState({
+    _id: '',
+    day: '',
+    hour: '',
+    activityId: '',
+    trainerId: '',
+    slots: ''
+  });
+
+  const onChangeInput = (e) => {
+    setKlass({
+      ...klass,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const funcionCargaDatos = (item) => {
+    setKlass({
+      _id: item._id,
+      day: item.day,
+      hour: item.hour,
+      activityId: item.activityId,
+      trainerId: item.trainerId,
+      slots: item.slots
+    });
+    console.log(item);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onEditItem(klass._id, klass);
+    console.log(klass);
+    setKlass({
+      day: '',
+      hour: '',
+      activityId: '',
+      trainerId: '',
+      slots: ''
+    });
+  };
+
+  return (
+    <div>
+      <table>
+        <thead>
+          <tr>
+            <th>Activity Name</th>
+            <th>day</th>
+            <th>hour</th>
+            <th></th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((item) => {
+            return (
+              <tr key={item._id}>
+                {item.activityId.name ? <td>{item.activityId.name}</td> : <td>Undefined</td>}
```

Comment:
is this necessary?

Reply:
Removed due to not being completly relevant.

It was ment to be part of a validation, but I focused on other parts and ended up incomplete.